src/commands/timeout.ts

```diff
@@ -0,0 +1,160 @@
+//Template: https://github.com/Costpap/CostBot/blob/main/src/commands/kick.ts
+
+//imports
+import { ChatInputCommandInteraction, PermissionFlagsBits, SlashCommandBuilder } from 'discord.js';
+import { notifyUser } from '../utils/misc';
+import ms from 'ms';
+
+export default {
+    data: new SlashCommandBuilder()
+        .setName('timeout')
+        .setDescription('Manages the timeout of a selected user.')
+        .addSubcommand((subcommand) =>
+            subcommand
+                .setName('add')
+                .setDescription('Times out the selected user.')
+                .addUserOption((option) => option.setName('user').setDescription('User to time out').setRequired(true))
+                .addStringOption((option) =>
+                    option
+                        .setName('duration')
+                        .setDescription('Duration of the timeout (e.g. 10sec/60m/2.5 hrs)')
+                        .setRequired(true),
+                )
+                .addStringOption((option) =>
+                    option.setName('reason').setDescription('Reason for timing out the user').setRequired(false),
+                )
+                .addBooleanOption((option) =>
+                    option.setName('silent').setDescription('Whether or not to DM the user').setRequired(false),
+                ),
+        )
+        .addSubcommand((subcommand) =>
+            subcommand
+                .setName('remove')
+                .setDescription('Removes timeout from a user.')
+                .addUserOption((option) =>
+                    option.setName('user').setDescription('User to remove the timeout from').setRequired(true),
+                )
+                .addStringOption((option) =>
+                    option.setName('reason').setDescription('Reason for removing the timeout').setRequired(false),
+                ),
+        ),
+    run: async (interaction: ChatInputCommandInteraction) => {
+        // Typeguard in order to ensure having access to ChatInputCommand interaction options.
+        if (!interaction.isChatInputCommand()) return;
+
+        if (interaction.inGuild() === false) {
+            return interaction.reply({ content: "❌ I can't execute this command inside DMs!", ephemeral: true });
+        }
+
+        if (!interaction.guild.members.me.permissions.has(PermissionFlagsBits.ModerateMembers)) {
+            return interaction.reply({
+                content: '❌ Sorry, I need the `Moderate Members` in order to execute this command.',
+                ephemeral: true,
+            });
+        }
+        if (typeof interaction.member.permissions === 'string')
+            return interaction.reply({ content: '❌ Unknown permissions. Please try again later.', ephemeral: true });
+        else if (!interaction.member.permissions.has(PermissionFlagsBits.ModerateMembers)) {
+            return interaction.reply({
+                content: '⛔ You need the `Moderate Members` permission in order to use this command!',
```

Comment:
Please change `Moderate Members` to `Timeout Members`, in accordance with what is shown in Discord's UI.
```suggestion
                content: '⛔ You need the `Timeout Members` permission in order to use this command!',
```

Reply:
Same as prevoiusly written.
![image](https://user-images.githubusercontent.com/90683098/193908019-8c9684b5-6210-4938-85e2-64b96efb11aa.png)